### PR TITLE
Better support for command-line snippet highlighting

### LINKED ISSUE #466: Support better highlighting of command-line snippets
From @kwalrath's comment in https://github.com/dart-lang/site-www/pull/463#pullrequestreview-82017078:

> fix the formatting of the command-line snippets? "Wrote data for Han Solo." is unnecessarily colorized.

Here is the snippet:
screen shot 2017-12-09 at 13 23 38

which is rendered from

```
   {% prettify bash %}
   $ cd httpserver/bin
   $ dart basic_writer_client.dart
   Wrote data for Han Solo.
   {% endprettify %}
```

---

For one thing, the prettify plugin could be modified to support `nocode`.

Or, we could fallback to using markdown fenced code (with `nocode`), which will fall back to using [Rouge](https://github.com/jneen/rouge). The default `<pre>` styling results in:

screen shot 2017-12-09 at 13 51 25

This is fine, but we'll need to change all command-line snippets.

We can actually do a better. Rouge supports `terminal`, which combined with some custom styling can yield something like this:

screen shot 2017-12-09 at 13 53 57

We could get similar results with prettify if we write a custom command-line / terminal lexer.

At this point, I'd vote for using Rouge's `terminal` formatting + custom styles

@kwalrath WDYT?

### PR DESCRIPTION
**Pull request by chalin**
Fixes #466 

Staged at https://dartlang-org-staging-0.firebaseapp.com, in particular see

- https://dartlang-org-staging-0.firebaseapp.com/tutorials/dart-vm/httpserver
- https://dartlang-org-staging-0.firebaseapp.com/tutorials/dart-vm/cmdline
- https://dartlang-org-staging-0.firebaseapp.com/install/linux
- https://dartlang-org-staging-0.firebaseapp.com/install/mac
- https://dartlang-org-staging-0.firebaseapp.com/install/windows

---

### Example

Styling updated from this:

> screen shot 2017-12-12 at 12 19 17

To this:

> screen shot 2017-12-12 at 12 19 47

### CODE REVIEW

File: src/_tutorials/dart-vm/cmdline.md [hunk 1]
```diff
@@ -78,14 +78,12 @@ void main() {
 
 <li markdown="1">
 In the directory that contains the file you just created,
-run the program with the command as shown by the highlighted text.
+run the program with the command as shown next.
```
Comment (kwalrath) [inline src/_tutorials/dart-vm/cmdline.md]: Let's change this to:

... run the program:

(I don't think the words afterward add anything.)

File: src/_tutorials/dart-vm/httpserver.md [hunk 2]
```diff
@@ -98,11 +98,11 @@ with the string `Hello, world!`
 At the command line, run the `hello_world_server.dart` script.
 You will see the following:
```
Comment (kwalrath) [inline src/_tutorials/dart-vm/httpserver.md]: Let's delete the second sentence, changing this to:

...script:

File: src/_tutorials/dart-vm/httpserver.md [hunk 3]
```diff
@@ -527,24 +527,24 @@ Run the server and client on the command line.
 
 1. First the server:
```
Comment (kwalrath) [inline src/_tutorials/dart-vm/httpserver.md]: Let's change this to:

First, run the server:

File: src/_tutorials/dart-vm/httpserver.md [hunk 4]
```diff
@@ -527,24 +527,24 @@ Run the server and client on the command line.
...
 
-   {% prettify bash %}
+   ```terminal
    $ cd httpserver/bin
    $ dart basic_writer_server.dart
-   {% endprettify %}
+   ```
 
 2. Then, run the client in a new window:
```
Comment (kwalrath) [inline src/_tutorials/dart-vm/httpserver.md]: Let's change this to:

In a new window, run the client:

File: src/_tutorials/dart-vm/httpserver.md [hunk 5]
```diff
@@ -527,24 +527,24 @@ Run the server and client on the command line.
...
 
-   {% prettify bash %}
+   ```terminal
    $ cd httpserver/bin
    $ dart basic_writer_client.dart
    Wrote data for Han Solo.
-   {% endprettify %}
+   ```
 
 3. The server writes the data to `file.txt`:
```
Comment (kwalrath) [inline src/_tutorials/dart-vm/httpserver.md]: We need to change this, too. How about:

Look at the data that the server writes to `file.txt`:

File: src/_tutorials/dart-vm/cmdline.md (hunk 1 above)
Comment (chalin) [reply src/_tutorials/dart-vm/cmdline.md]: Ah, yes. Done.

File: src/_tutorials/dart-vm/httpserver.md (hunk 2 above)
Comment (chalin) [reply src/_tutorials/dart-vm/httpserver.md]: done

File: src/_tutorials/dart-vm/httpserver.md (hunk 3 above)
Comment (chalin) [reply src/_tutorials/dart-vm/httpserver.md]: done

File: src/_tutorials/dart-vm/httpserver.md (hunk 4 above)
Comment (kwalrath) [reply src/_tutorials/dart-vm/httpserver.md]: Yes, that's better. Thanks!

File: src/_tutorials/dart-vm/httpserver.md (hunk 5 above)
Comment (chalin) [reply src/_tutorials/dart-vm/httpserver.md]: Changed (with edits approved via chat).

**Pull request merged**